frontend/components/inputs/UploadTimeSeriesFileInput.js

```diff
@@ -1,99 +1,210 @@
-                min={constants[i][key]["min"]}
-                max={constants[i][key]["max"]}
-                step={constants[i][key]["step"]}
-                onChange={e => updateConstant(i, key, e.target.value)}
-                value={constants[i][key]["value"]}
-            />
-        </div>
-    );
+
+    const makeControl = (i, key) => {
+        return (
+            <div key={i + constantsDefaults[key]["label"]}>
+                <label>{constantsDefaults[key]["label"]}</label>
+                <input className="form-control my-3" type="number"
+                        min={constantsDefaults[key]["min"]}
+                        max={constantsDefaults[key]["max"]}
+                        step={constantsDefaults[key]["step"]}
+                        onChange={e => updateConstant(i, key, e.target.value)}
+                        value={constants[i][key]}
+                />
+            </div>
+        );
+    };
 
     return (
         <>
-            <FileInput onSubmit={submitFileToAPI} />
+            <div className={"time-series-input"}>
+                <label>Use every nth value</label>
+                <input className="form-control my-3" type="number"
+                        min={1}
+                        step={1}
+                        onChange={e => setEveryN(e.target.value)}
+                       value={everyN}
+                />
+            </div>
             <div className={"time-series-input"}>
                 <label>Duration of each step (sec)</label>
                 <input className="form-control my-3" type="number"
-                    min={0.1}
-                    max={10}
-                    step={.1}
-                    onChange={e => setDuration(e.target.value)}
-                    value={duration}
+                        min={0.1}
+                        max={10}
+                        step={.1}
+                        onChange={e => setDuration(e.target.value)}
+                       value={duration}
                 />
             </div>
+            <label>
+                <input type="checkbox" onClick={e => setMapToNote(e.target.checked)}/>
+                <span> Map numbers to note?</span>
+            </label>
+            <p>Column Constants:</p>
+            <button className="btn btn-secondary ml-1" onClick={addColumn}>+</button>
+            <button className="btn btn-secondary ml-1" onClick={removeColumn}>-</button>
             <div className="form-inline">
                 {constants.map((col, i) => {
                     return <div className={"time-series-input"} key={i}>
-                        <p>Column {i}: </p>
+                        <p>Column {i+1}: </p>
                         {Object.keys(col).map(constant => {
                             return makeControl(i,constant);
                         })}
                     </div>;
                 })}
             </div>
+            <form className="form-inline">
```

Comment:
It looks like this PR undoes work from #49 that factored this code out. I'm going to go ahead and repair myself, but moving forward please make sure to update your working branch from `main` regularly and to be sensitive to work that's going on in other branches when you make a PR.